gulpfile.coffee

```diff
@@ -54,11 +57,20 @@ gulp.task 'lint', ->
 gulp.task 'build', (callback) ->
 	runSequence('lint', 'clean-build', 'build-node', 'build-browser', callback)
 
-gulp.task 'build-node', ->
+
+gulp.task 'build-node-ts', ->
+	tsProject = typescript.createProject('tsconfig.json')
+	gulp.src(OPTIONS.files.tsAppFiles)
+		.pipe(tsProject()) .on('error', gutil.log)
+		.pipe(gulp.dest(OPTIONS.directories.build))
+
+gulp.task 'build-node-coffee', ->
 	gulp.src(OPTIONS.files.app)
 		.pipe(coffee(header: true, bare: true)).on('error', gutil.log)
 		.pipe(gulp.dest(OPTIONS.directories.build))
 
+gulp.task 'build-node', ['build-node-ts', 'build-node-coffee']
```

Comment:
In general, I'd prefer to start moving away from gulp entirely - it's not making this any easier, and it's an annoy extra layer (or multiple layers) between you and what's going on, which is fairly annoying to work with imo. You can do all of this change except the mocha bit for example by just swapping `gulp build` in package.json for `tsc && gulp build`, which is way easier, and has a lot fewer moving parts.

Reply:
Didn't have that in mind and thought that I should
* extend instead of write something that would fragment the build logic in two places
* make the gulp tasks parellelable

Moving some of the gulp tasks to package.json as you suggested.